source/postprocess/visualization.cc

```diff
@@ -235,12 +235,18 @@ namespace aspect
     template <class Archive>
     void Visualization<dim>::OutputHistory::serialize (Archive &ar, const unsigned int)
     {
-      ar &mesh_changed
+      ar
       & last_mesh_file_name
       & times_and_pvtu_names
       & output_file_names_by_timestep
       & xdmf_entries
       ;
+
+      // We do not serialize mesh_changed but use the default (true) from our
+      // constructor. This will result in a new mesh file the first time we
+      // create visualization output after resuming from a snapshot. Otherwise
+      // we might get corrupted graphical output, because the ordering of
+      // vertices can be different after resuming.
```

Comment:
I had to think about this for a good long time. Can you add something like
```suggestion
      // vertices can be different after resuming. (This is because when
      // deal.II builds a triangulation, it tears down and re-creates its mesh every time
      // p4est changes the partitioning; this process introduces a history where
      // cells and vertices may be numbered differently if the triangulation
      // had previous content than when -- as is done after a restart -- the
      // triangulation is empty.)
```